test/e2e/flowaggregator_test.go

```diff
@@ -1510,17 +1515,24 @@ func getClickHouseOutput(t *testing.T, data *TestData, srcIP, dstIP, srcPort str
 	return flowRecords
 }
 
-func getRecordsFromOutput(t *testing.T, output, labelFilter string) []string {
-	re := regexp.MustCompile("(?m)^.*" + "#" + ".*$[\r\n]+")
-	output = re.ReplaceAllString(output, "")
-	output = strings.TrimSpace(output)
-	recordSlices := strings.Split(output, "IPFIX-HDR:")
-	if labelFilter == "" {
-		return recordSlices
+func getRecordsFromOutput(t *testing.T, output, labelFilter, src, dst, srcPort string) []string {
+	var response IPFIXCollectorResponse
+	err := json.Unmarshal([]byte(output), &response)
+	if err != nil {
+		require.NoErrorf(t, err, "error when unmarshall output from IPFIX collector Pod")
 	}
+	recordSlices := response.FlowRecords
 	records := []string{}
 	for _, recordSlice := range recordSlices {
-		if strings.Contains(recordSlice, labelFilter) {
+		// We don't check the last record.
+		if strings.Contains(recordSlice, "octetDeltaCount: 0") {
```

Comment:
What's the purpose for skipping the last record?

Reply:
Previously, we were checking octetDeltaCount != 0 for all the records (except the last one). The delation for querying records from the IPFIX collector (solved) can make the last record appear in the collector. I think it is much safer to filter out the last one because we don't verify any information for it.